test/test_cuda.py

```diff
@@ -2521,6 +2521,20 @@ def _grad_scaling_autocast_fused_optimizers(self, optimizer_ctor, optimizer_kwar
                         actual = actual.squeeze()
                     self.assertEqual(state_control[k], actual)
 
+    def test_grad_scale_will_not_overflow(self):
+        model = torch.nn.Linear(5, 1).cuda()
+        optimizer = torch.optim.Adam(model.parameters())
+        scaler = torch.cuda.amp.GradScaler(growth_interval=1)
+        for _ in range(150):
+            optimizer.zero_grad()
+            x = torch.randn(1, 5).cuda()
+            y = 1e-30 * torch.randn(1, 1).cuda()
+            l = ((model(x) - y)**2).mean()
+            scaler.scale(l).backward()
+            scaler.step(optimizer)
+            scaler.update()
+        assert(scaler._scale is not float('inf') and scaler._scale is not float('nan'))
```

Comment:
Hmm, 150 iterations sound lot while the computation is quite light.
What do you think about initializing `GradScaler` with a ridiculously large `growth_factor` and `growth_interval=1` so that only one `scaler.step` would cause overflow?

ref: https://pytorch.org/docs/stable/amp.html#torch.cuda.amp.GradScaler

Reply:
I've contrived something similar 😛